useragent/user_agent.go

```diff
@@ -79,19 +86,48 @@ type info struct {
 	Value string
 }
 
+// Sanitize the user agent value. This is useful when the value is not ensured to be
+// to be valid at compile time. Having this sanitization then ensures downstream
+// applications can correctly parse the full user agent header, by making sure
+// characters like '/' and ' ' are not present in the value.
+func Sanitize(s string) string {
+	allowList := func(r rune) rune {
+		if r == '.' || unicode.IsLetter(r) || unicode.IsDigit(r) {
+			return r
+		}
+		return '-'
+	}
+
+	return strings.Map(allowList, s)
+}
+
 func (u info) String() string {
 	return fmt.Sprintf("%s/%s", u.Key, u.Value)
 }
 
 type data []info
 
+// Validate the key value pair being set in the user agent. Error if invalid.
+func validate(key, value string) error {
+	// DBR versions as set in the `DATABRICKS_RUNTIME_VERSION` environment variable
+	// are not valid semver strings. Eg: 15.1 or client.0. Thus we allow arbitrary
+	// values for the runtime key.
```

Comment:
We could broaden `regexpAlphanum` class to include `.`. That seems less brittle than the current approach.

Are there other examples of values that we need to allow?

Reply:
Done. We don't need any other characters at the moment, but we can expand this set as and when needed.